File: src/Java/Article.java
package Java;

import java.util.ArrayList;
import java.util.List;

public class Article extends Post {
    private String editorialStatus; // sent, aproved, rejected
    private static int coutsId;;
    private static int coutsCommentsId;
    private int id;

    public Article(String title, String content, User author){
        super(title, content, author);
        this.id = ++Article.coutsId;
        this.editorialStatus = "sent";
        
        this.comments = new ArrayList<>();
    }

    // Metodos Getters
    public String getEditorialStatus(){
        return this.editorialStatus;
    }

    public List<Comment> getComments(){
        return this.comments;
    }

    public int getId(){
        return this.id;
    }

    // Metodos Setters
    public void setEditorialStatus(String editorialStatus){
        this.editorialStatus = editorialStatus;
    }

    @Override
    public String toString(){
        StringBuilder sb = new StringBuilder();
        sb.append(super.toString());
        sb.append("Id: " + this.id + "\n");
        sb.append("Estado de editorial: " + this.editorialStatus + "\n");
        if (super.comments.size() > 0){
            sb.append("-------------------------------------\n");
            sb.append("Comentarios: \n");
            for(Comment c : super.comments){
                sb.append(c);
                sb.append("\n");
            }
        } else {
            sb.append("No hay comentarios\n");
        }
        sb.append("-------------------------------------");
        return sb.toString();
    }


    @Override
    public void addComment(String text, User author){
        this.comments.add(new Comment(++Article.coutsCommentsId, text, author, this));
        Notification newNotification = new Notification("Nuevo Comentario", this.author.getEmail(), this.author.getName(), this.author.getPhoneNumber(), "Nuevo comentario de "+author.getUsername()+ "\nComentario: \n"+ text);
        this.author.getNotifications().add(newNotification);
    }
}

File: src/C#/networking-app/MyApplication/Article.cs

using System.Text;

namespace MyApplication.Models
{
	public class Article : Post
	{
		private string EditorialStatus { get; set; } // "sent", "approved", "rejected"
		private static int coutsId;
		private int id;
		private static int coutsCommentsId;

		// Constructor
		public Article(string title, string content, User author) : base(title, content, author)
		{	
			this.id = ++Article.coutsId;
			EditorialStatus = "sent";
			this.comments = new List<Comment>();
		}

		// Métodos Getters y Setters
		public string GetEditorialStatus() => EditorialStatus;

		public void SetEditorialStatus(string editorialStatus) => EditorialStatus = editorialStatus;
		
		public int GetId() => this.id;
		
		public void SetId(int id) => this.id = id;
		
		public List<Comment> GetComments() => this.comments;
		
		public override string ToString(){
			StringBuilder sb = new StringBuilder();
			sb.Append(base.ToString());
			sb.Append("Id: " + this.id + "\n");
			sb.Append("Estado de editorial: " + this.EditorialStatus + "\n");
			if (base.comments.Count > 0)
			{
				sb.Append("-------------------------------------\n");
				sb.Append("Comentarios: \n");
				foreach (Comment c in base.comments)
				{
					sb.Append(c.ToString());
					sb.Append("\n");
				}
			}
			else
			{
				sb.Append("No hay comentarios\n");
			}
			sb.Append("-------------------------------------");
			return sb.ToString();
		}
		
		
		public override void AddComment(string text, User author){
			this.comments.Add(new Comment(++Article.coutsCommentsId, text, author, this));
			Notification newNotification = new Notification("Nuevo Comentario","Nuevo comentario de "+author.Username+ "\nComentario: \n"+ text ,this.author.Email, this.author.Name, this.author.PhoneNumber); 
			this.author.Notifications.Add(newNotification);
		}
	}

}
